### ISSUE TITLE
Build system improvements

### ISSUE THREAD

**Issue by joakim-tjernlund** [clhedrick/kerberos#1]
I just found this project and I think it can be useful in our site.

I note however that the build system is plain Makefile's with hardcoded paths and
tools (like gcc).
This makes it hard to build/deploy in a standard way.
Any plans to modernize the build system(using autotools, cmake or meson)?

**Comment by joakim-tjernlund** [comment]
Ping?

**Comment by clhedrick** [comment]
No. It's a good idea, but at the moment we only need one binary for all of our systems. (I recently did a change because we inadvertently broke backwards compatibility to old releases.) I'd be happy to accept contributed changes. 

I should note that I'm working with upstream maintainers to get rid of the need for a lot of this. sssd should be able to renew credentials. Other parts are hacks to get around problems with individual tools that I'd love to see patched there. But my suggestions don't seem to be getting much response.

The one tool that can't be absorbed in an obvious way is credserv / kgetcred.

**Comment by joakim-tjernlund** [comment]
Sorry, but I am not working on build improvements. I found this and figured I would take it for a spin but backed away when I could not install/use it without substitutional hacking. It was just to much for a test drive. Sorry

My experience is that upstream move very slowly so it will be a long journey.

**Comment by clhedrick** [comment]
If you're interested in a fairly recent Linux (Ubuntu since 14, 16, 18, Centos 6, 7, SLES 12), you can use my binaries. They're been tested there. Check out the kerberos-ansible repo. The files are all in kerberos-files. Most of the binaries are in kerberos-files/copy.  I build on Centos 7,  but take precautions to make sure that the binaries will work on all those systems.

See README for how the ansible scripts are intended to be used.

**Comment by joakim-tjernlund** [comment]
I am on Gentoo, a rolling, build from src distro, so precompiled binaries from other distros will not work :(

**Comment by clhedrick** [comment]
If the same binary works on Centos, Ubuntu and SLES, why wouldn't it work with Gentoo? It depends upon basic GCC and Kerberos libraries.

**Comment by joakim-tjernlund** [comment]
> If the same binary works on Centos, Ubuntu and SLES, why wouldn't it work with Gentoo? It depends upon basic GCC and Kerberos libraries.

Sorry, I completely forgot about this issue.

That would imply I would always have to run a binary pkg and that will not work in the long run so
there would be little point to test it out then, sorry.
